cluster-provision/gocli/opts/featuregate/featuregate.go

```diff
@@ -0,0 +1,150 @@
+package featuregate
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"kubevirt.io/kubevirtci/cluster-provision/gocli/pkg/libssh"
+)
+
+type featureGateOpt struct {
+	sshClient    libssh.Client
+	featureGates []string
+}
+
+type component string
+
+const (
+	componentKubeAPIServer     component = "kube-apiserver"
+	componentKubeControllerMgr component = "kube-controller-manager"
+	componentKubeScheduler     component = "kube-scheduler"
+
+	searchFeatureGatesInFileFormat                    = "awk '/feature-gates/' %s"
+	getComponentCommandFormat                         = "kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods -n kube-system -l component=%s -o jsonpath='{.items[0].spec.containers[*].command}'"
+	getComponentReadyContainersFormat                 = "kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods -n kube-system -l component=%s -o=jsonpath='{.items[0].status.containerStatuses[*].ready}'"
+	addFeatureGatesToComponentCommandFormat           = `sudo sed -i 's/- %s/- %s\n    - %s/g' /etc/kubernetes/manifests/%s.yaml`
+	searchComponentsFilesCommand                      = "ls /etc/kubernetes/manifests"
+	addFeatureGatesFieldToKubeletConfigCommand        = "sudo /bin/su -c \"echo -e 'featureGates:' >> /var/lib/kubelet/config.yaml\""
+	addFeatureGatesToKubeletConfigCommandFormatFormat = `sudo sed -i 's/featureGates:/featureGates:\n  %s: true/g' /var/lib/kubelet/config.yaml`
+	kubeletRestartCommand                             = "sudo systemctl restart kubelet"
+
+	featureGateExistInKubeletError                = "feature gates should not exist in kubelet by default"
+	featureGateExistInComponentCommandErrorFormat = "feature gates should not exist in %v command by default"
+)
+
+func NewFeatureGatesOpt(sc libssh.Client, featureGates []string) *featureGateOpt {
+	return &featureGateOpt{
+		sshClient:    sc,
+		featureGates: featureGates,
+	}
+}
+
+func (o *featureGateOpt) Exec() error {
+	output, err := o.sshClient.CommandWithNoStdOut(searchFeatureGatesInFile("/var/lib/kubelet/config.yaml"))
+	if err != nil {
+		return err
+	}
+	fgsExist := len(output) > 0
+	if fgsExist {
+		return fmt.Errorf(featureGateExistInKubeletError)
+	}
```

Comment:
Is it possible to add features in this case / override the existing ones?
However this shouldn't block this PR.

Reply:
By default, no feature gates should be enabled in KubeVirt CI. This is just a sanity check to ensure that no one is modifying the feature gates in earlier stages.

Reply:
Done, now we can disable featureGates as well :)